services/dashboard/views/restapi.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0
+ * See accompanying LICENSE file.
+ */
+'use strict';
+angular.module('dashboard.restapi', [])
+
+  .factory('restapi', ['$http', '$timeout', 'conf', function ($http, $timeout, conf) {
+    return {
+      /** Get data from server periodically before a scope is destroyed. */
+      subscribe: function (url, scope, onSuccess, onError) {
+        var timeoutPromise;
+        scope.$on('$destroy', function () {
+          $timeout.cancel(timeoutPromise);
+        });
+        var fn = function () {
+          var cancel = false;
```

Comment:
Here we are doing polling via GET. I think a better approach is to establish a refresh interval on the server where it will push any new data via websockets. So we would need to add this refresh interval into the websocket initial request. The server would then set an akka scheduler so this information is sent in these intervals. If no new data is available nothing should be sent. This removes needless polling from client and will allow client to scale when we are sending lots of data. If you would like to handle this as a follow up issue - make a note here in a //TODO, something like
//TODO convert to websocket push model See <issueid>.